### PR TITLE
fetch-recording (#23503)

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-3845

## Description
rewrite the fetch recording command
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 5.0.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by ShirleyDenkberg** [comment]
@JasBeilin @ilaner Doc review completed.

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 1]
```diff
@@ -141,13 +144,14 @@ def get_oauth_token(self, force_gen_new_token=False):
...
-        ctx.update({'token_info': {'oauth_token': oauth_token, 'generation_time': now.strftime("%Y-%m-%dT%H:%M:%S")}})
+        ctx.update({'token_info': {'oauth_token': oauth_token,
+                   'generation_time': now.strftime("%Y-%m-%dT%H:%M:%S")}})
         set_integration_context(ctx)
         return oauth_token
 
     def error_handled_http_request(self, method, url_suffix='', full_url=None, headers=None,
                                    auth=None, json_data=None, params=None,
-                                   return_empty_response: bool = False, resp_type: str = 'json'):
+                                   return_empty_response: bool = False, resp_type: str = 'json', stream: bool = False):
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:154]: What is the "stream" arg? Lets give it a meaningful name.

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 2]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
+    meeting_id = args.get('meeting_id')
+    delete_after = argToBoolean(args.get('delete_after'))
+    client = client
+
+    data = client.zoom_fetch_recording(
+        method='GET',
+        url_suffix=f'meetings/{meeting_id}/recordings'
+    )
+    recording_files = data.get('recording_files')
+    # getting the audio and video files
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:621]:
```suggestion
    # Getting the audio and video files which are contained in every recording.
```

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 3]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
+    for file in recording_files:
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:622]: Is is possible to have no recording? If so, do we want to have a better notice to the client?

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 4]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
+        download_url = file.get('download_url')
+        try:
+            # download the file
+            demisto.debug(
+                f"Trying to download the files of meeting {meeting_id}")
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:627]:
```suggestion
            demisto.debug(f"Trying to download the files of meeting {meeting_id}.")
```

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 5]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
+            record = client.zoom_fetch_recording(
+                method='GET',
+                full_url=download_url,
+                resp_type='response',
+                stream=True
+            )
+            # save the file
+            filename = f'recording_{meeting_id}_{file.get("id")}.mp4'
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:635]: Both audio and video are mp4 files? Have you tried downloading and checking the downloaded files are legit?

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 6]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
-            rf = requests.delete(URL + 'meetings/%s/recordings/%s' % (meeting, file['id']), headers=HEADERS,
-                                 params=PARAMS, verify=USE_SSL)
-            if rf.status_code == 204:
-                demisto.results('File ' + filename + ' was moved to trash.')
-            else:
-                demisto.results('Failed to delete file ' + filename + '.')
-    else:
-        return_error('Download of recording failed: [%d] - %s' % (res.status_code, res.text))
+                record.raw.decode_content = True
+                shutil.copyfileobj(record.raw, f)
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:638]:
```suggestion
                # Saving the content of the file locally.
                record.raw.decode_content = True
                shutil.copyfileobj(record.raw, f)
```

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 7]
```diff
@@ -582,61 +606,72 @@ def zoom_create_meeting_command(client, **args) -> CommandResults:
...
+                try:
+                    # delete the file from the cloud
+                    demisto.debug(f"Trying to delete the file {filename}")
+                    client.zoom_fetch_recording(
+                        method='DELETE',
+                        url_suffix=f'meetings/{meeting_id}/recordings/{file["id"]}',
+                        resp_type='response'
+                    )
+                    results.append(CommandResults(
+                        readable_output=f"The file {filename} was successfully removed from the cloud"))
```
Comment (JasBeilin) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:654]:
```suggestion
                        readable_output=f"The file {filename} was successfully removed from the cloud."))
```

**Review by JasBeilin — CHANGES_REQUESTED** [review]
Nice work. Please check my comments. Also:
1. Please make sure your issue is connected to this PR.
2. Please notice that there are many changes here (flake8) that you did not make I assume. This is causing the PR to contain a lot more unnecessary changes.
3. Lets schedule a short demo for this command after you finish adding all your changes.

File: Packs/Zoom/.pack-ignore [hunk 8]
```diff
@@ -1,5 +1,6 @@
 [file:Zoom_IAM.yml]
-ignore=BA108,BA109
```
Comment (ilaner) [inline Packs/Zoom/.pack-ignore:2]: please revert

File: Packs/Zoom/.pack-ignore [hunk 9]
```diff
@@ -1,5 +1,6 @@
...
+ignore = BA108, BA109
 
 [file:Zoom_IAM_image.png]
-ignore=IM111
```
Comment (ilaner) [inline Packs/Zoom/.pack-ignore:5]: this too

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 10]
```diff
@@ -88,10 +88,12 @@ def __init__(
         self.account_id = account_id
         self.client_id = client_id
         self.client_secret = client_secret
-        is_jwt = (api_key and api_secret) and not (client_id and client_secret and account_id)
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:91]: revert

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 11]
```diff
@@ -88,10 +88,12 @@ def __init__(
...
+        is_jwt = (api_key and api_secret) and not (
+            client_id and client_secret and account_id)
         if is_jwt:
             # the user has chosen to use the JWT authentication method (deprecated)
-            self.access_token: str | None = get_jwt_token(api_key, api_secret)  # type: ignore[arg-type]
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:94]: revert

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 12]
```diff
@@ -407,7 +424,8 @@ def zoom_list_users_command(client, **args) -> CommandResults:
 
             md = tableToMarkdown('Users', minimal_needed_info, ['id', 'email',
                                                                 'type', 'pmi', 'verified', 'created_at', 'status', 'role_id'])
-            md += '\n' + tableToMarkdown('Metadata', [raw_data][0][0], ['total_records'])
+            md += '\n' + tableToMarkdown('Metadata',
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:427]: revert

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 13]
```diff
@@ -761,16 +802,18 @@ def main():  # pragma: no cover
...
-            zoom_fetch_recording_command()
+            results = zoom_fetch_recording_command(client, **args)
         elif command == 'zoom-list-users':
             results = zoom_list_users_command(client, **args)
         else:
             return_error('Unrecognized command: ' + demisto.command())
         return_results(results)
 
     except DemistoException as e:
+        if e.res:
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:813]: seems like a bug?

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 14]
```diff
@@ -761,16 +802,18 @@ def main():  # pragma: no cover
...
+            return_results(e.res)
         # For any other integration command exception, return an error
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:815]: lets add the traceback in the debug logs

File: Packs/Zoom/Integrations/Zoom/Zoom.py [hunk 15]
```diff
@@ -718,8 +756,10 @@ def main():  # pragma: no cover
     params = demisto.params()
     args = demisto.args()
     base_url = params.get('url')
-    api_key = params.get('creds_api_key', {}).get('password') or params.get('apiKey')
-    api_secret = params.get('creds_api_secret', {}).get('password') or params.get('apiSecret')
+    api_key = params.get('creds_api_key', {}).get(
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.py:759]: revert

**Review by ilaner — CHANGES_REQUESTED** [review]
Looks good.
Please revert all the formatting changes, and see the comments

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 1 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:154]: thats the original name of the argument in the http_request, all i did is to pick out only the argument im using. so i think the best is to keep the name.

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 3 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:622]: that is possible. the error will be raised bi the first call to the API (L616)

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 5 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:635]: yes. mp4a is for audio. i tried many times and it works like a charm.

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 15 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:759]: this is a fix @MichaelYochpaz did, it's already morged.

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 10 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:91]: this is a fix @MichaelYochpaz did, it's already morged. Not my code.

File: Packs/Zoom/Integrations/Zoom/Zoom.py (hunk 11 above)
Comment (RosenbergYehuda) [reply Packs/Zoom/Integrations/Zoom/Zoom.py:94]: this is a fix @MichaelYochpaz did, it's already morged. Not my code.

**Review by JasBeilin — APPROVED** [review]

**Review by JasBeilin — APPROVED** [review]

File: Packs/Zoom/Integrations/Zoom/Zoom.yml [hunk 16]
```diff
@@ -1,6 +1,6 @@
 category: Messaging
 commonfields:
-  id: Zoom
+  id: Zoom Dev
```
Comment (ilaner) [inline Packs/Zoom/Integrations/Zoom/Zoom.yml:3]: Revert to zoom

**Review by ilaner — CHANGES_REQUESTED** [review]
see last comment

**Review by ilaner — APPROVED** [review]
Great work

**Pull request merged**